test/commands_test.py

```diff
@@ -0,0 +1,211 @@
+        )
+
+    @responses.activate
+    def test_projects_names_only(self, test_connection, capsys):
+        responses.add(
+            responses.GET,
+            test_connection.get_route("get_projects"),
+            json={
+                "projects": [
+                    {"archived_at": "some datetime", "id": "p123", "name": "Foo"}
+                ]
+            },
+        )
+
+        ret = commands.projects(test_connection, names_only=True)
+
+        expected = {"p123": "Foo"}
+        assert ret == expected
+        captured = capsys.readouterr()
+        assert captured.out == f"{expected}\n"
+
+    @responses.activate
+    def test_projects_deprecated_i(self, test_connection, capsys):
+        responses.add(
+            responses.GET,
+            test_connection.get_route("get_projects"),
+            json={
+                "projects": [
+                    {"archived_at": "some datetime", "id": "p123", "name": "Foo"}
+                ]
+            },
+        )
+
+        with pytest.warns(FutureWarning):
+            ret = commands.projects(test_connection, i=True)
+
+        expected = {"p123": "Foo"}
+        assert ret == expected
+        captured = capsys.readouterr()
+        assert captured.out == f"{expected}\n"
+
+    @responses.activate
+    def test_projects_json_flag(self, test_connection, capsys):
+        responses.add(
+            responses.GET,
+            test_connection.get_route("get_projects"),
+            json={
+                "projects": [
+                    {"archived_at": "some datetime", "id": "p123", "name": "Foo"}
+                ]
+            },
+        )
+
+        ret = commands.projects(test_connection, json_flag=True)
+
+        expected = [{"archived_at": "some datetime", "id": "p123", "name": "Foo"}]
+        assert ret == expected
+        captured = capsys.readouterr()
+        assert captured.out == f"{json.dumps(expected)}\n"
+
+    @responses.activate
+    def test_projects_default_return(self, test_connection, capsys):
+        responses.add(
+            responses.GET,
+            test_connection.get_route("get_projects"),
+            json={
+                "projects": [
+                    {"archived_at": "some datetime", "id": "p123", "name": "Foo"}
+                ]
+            },
+        )
+
+        commands.projects(test_connection)
+
+        captured = capsys.readouterr()
+        assert captured.out == (
+            "\n"
+            "                                   PROJECTS:\n"
+            "                                   \n"
+            "              PROJECT NAME              |               PROJECT ID               \n"
+            "--------------------------------------------------------------------------------\n"
+            "Foo (archived)                          |                  p123                  \n"
+            "--------------------------------------------------------------------------------\n"
+        )
+
+
+class TestSubmit:
```

Comment:
Wondering if we want to move `Submit` tests into a separate test file.

Reply:
Sure! Yea, I'll reorg it to break these into separate files